src/components/metadataEditor/metadataEditor.js

```
@@ -38,35 +38,33 @@ function closeDialog() {
     }
 }
 
-function submitUpdatedItem(form, item) {
-    function afterContentTypeUpdated() {
-        toast(globalize.translate('MessageItemSaved'));
+async function submitUpdatedItem(form, item) {
+    const apiClient = getApiClient();
 
+    const afterContentTypeUpdated = () => {
+        toast(globalize.translate('MessageItemSaved'));
         loading.hide();
         closeDialog();
-    }
-
-    const apiClient = getApiClient();
-
-    apiClient.updateItem(item).then(function () {
-        const newContentType = form.querySelector('#selectContentType').value || '';
+    };
 
-        if ((metadataEditorInfo.ContentType || '') !== newContentType) {
-            apiClient.ajax({
+    try {
+        await apiClient.updateItem(item);
+        const newContentType = form.querySelector('#selectContentType')?.value || '';
 
-                url: apiClient.getUrl('Items/' + item.Id + '/ContentType', {
+        if ((metadataEditorInfo?.ContentType || '') !== newContentType) {
```

We probably shouldn't even start doing anything if `metadataEditorInfo` is undefined/null. Maybe just throw an exception at the beginning of the function instead of optional chaining?

---

seeing that this check is to just update the items content type when there is change in content type and ignore if otherwise. we can directly ignore the check and lets do the update content api even if its redundant.